distutils/ccompiler.py

```diff
@@ -802,7 +802,10 @@ def has_function(self, funcname, includes=None, include_dirs=None,
         except (LinkError, TypeError):
             return False
         else:
-            os.remove("a.out")
+            output_dir = self.output_dir
+            if output_dir is None:
+                output_dir = ''
+            os.remove(os.path.join(output_dir, "a.out"))
```

Comment:
```suggestion
            os.remove(os.path.join(self.output_dir or '', "a.out"))
```

Reply:
My code follows the handling of `None` in other places in the file.

Reply:
Yeah, this is just a suggestion. It is purely stylistic. I'll leave it up to Jason.

Reply:
Both good suggestions. I like FFY00's suggestion, even though it's inconsistent. Since distutils is old code, I'll err on the side of using better syntax where possible.